extensions/wikia/EmergencyBroadcastSystem/EmergencyBroadcastSystemController.class.php

```diff
@@ -2,8 +2,9 @@
 class EmergencyBroadcastSystemController extends WikiaController {
 
 	public function index() {
-		if ( $this->isCorrectPage() && $this->isPowerUser() && $this->hasNonPortableInfoBoxes() && $this->canOpenEBS() ) {
-			$this->response->setVal( 'nonPortableCount', '3' ); // Temporary number for testing
+		$nonPortableCount = $this->countOfNonPortableInfoBoxes();
+		if ( $this->isCorrectPage() && $this->isPowerUser() && $this->canOpenEBS() && ( $nonPortableCount > 0 ) ) {
```

Comment:
What about moving `( $this->isCorrectPage() && $this->isPowerUser() && $this->canOpenEBS() )` check before we make the `getNonportableInfoboxes()` query?